lib/gftp.h

```diff
@@ -494,14 +494,45 @@ struct gftp_request_tag
   SSL * ssl;
 #endif
 
-#if GLIB_MAJOR_VERSION > 1
   GIConv iconv_to, iconv_from; 
   unsigned int iconv_from_initialized : 1,
                iconv_to_initialized : 1;
   char *iconv_charset;
-#endif
 };
 
+/* Support deprecated functions for older glibs */
+#if GLIB_CHECK_VERSION(2,31,0)
+  #undef g_static_mutex_lock
+  #define g_static_mutex_lock g_mutex_lock
+  #undef g_static_mutex_unlock
+  #define g_static_mutex_unlock g_mutex_unlock
```

Comment:
I would just convert everything over to use the GMutex instead of the GStaticMutex. I don't think it is a good idea to redefine symbols in the glib library. Someone else looking through the code would expect one type of behavior, but doing things like this could cause it to have a different type of behavior, which will increase the over long-term maintenance of the project. Moving to GMutex should also get rid of all of the #ifdefs around the mutex code.